scanner/walk_dir_tree_test.go

```diff
@@ -43,36 +42,70 @@ var _ = Describe("load_tree", func() {
 
 	Describe("isDirOrSymlinkToDir", func() {
 		It("returns true for normal dirs", func() {
-			dir, _ := os.Stat("tests/fixtures")
-			Expect(isDirOrSymlinkToDir("tests", dir)).To(BeTrue())
+			dirents, _ := os.ReadDir("tests")
```

Comment:
This test does not test all cases: If the file is not found, the test will pass (as it won't execute the Expectation). Maybe a better approach is creating a helper function to be used as the previous `os.Stat`, something like:
```go
func getDirEntry(baseDir, name string) (fs.DirEntry, error) {
	dirEntries, _ := os.ReadDir(baseDir)
	for _, d := range dirEntries {
		if d.Name() == name {
			return d, nil
		}
	}
	return nil, os.ErrNotExist
}
```

Then use it in the test like this:
```go
It("returns false when folder name starts with ellipses", func() {
	dir, _ := getDirEntry(baseDir, "...unhidden_folder")
	Expect(isDirIgnored(baseDir, dir)).To(BeFalse())
})
```

Also notice that I used `baseDir` as some of the old tests were doing. This is actually required for the tests to pass in Windows. Looks like I was missing this in other tests, but while you are at it, can you please make sure we are doing the Right Thing ™️ ? Thanks!

Reply:
right on both counts; i just promoted to simpler way to both, i'll push back the harder way to all